File: src/main/java/org/gids/robot/speech/SpeechCallback.java
package org.gids.robot.speech;

/**
 * Grammar is defined in resources/org.gids.robot/grammars/command.gram
 */

public class SpeechCallback{
    public SpeechCallback(String result) {
        if(result.contains("hello")) {
            TTSService.getInstance().say("Hello Nikhil");
        } else if (result.matches("[A-Za-z]")) {
            TTSService.getInstance().say(result);
        }
    }
}

File: src/main/kotlin/org/gids/robot/speech/SpeechCallback.kt
package org.gids.robot.speech

import org.gids.robot.tts.TTSService
import java.util.*

/**
 * Grammar is defined in resources/org.gids.robot/grammars/command.gram
 */

class SpeechCallback(val result: String){
    val GREETINGS: Array<String> = arrayOf("hello", "hi")

    init {
        if(result.split(" ")[0] in GREETINGS) {
            TTSService.say("Hello Nikhil")
        } else {
            TTSService.say(result)
        }
    }
}
